python/cudf/cudf/core/frame.py

```diff
@@ -1322,7 +1322,7 @@ def fillna(self, value, method=None, axis=None, inplace=False, limit=None):
         copy_data = self._data.copy(deep=True)
 
         for name in copy_data.keys():
-            if name in value and value[name] is not None:
+            if name in value and value[name] is not None and value[name] is not cudf.NA:
```

Comment:
Maybe we should create a helper function for `is_scalar_null` because `is not None and is not cudf.NA` looks to be a repeated pattern.

Reply:
I actually think we can remove the check for `None` explicitly. As it stands after this is merged, we won't return `None` for nulls anymore. It would be `cudf.NA` always.

Reply:
@kkraus14 this does remind me of our discussion last thursday however: This PR would _not_ make it so that reductions or `__getitem__()` would yield cuDF scalars, but it *does* make it so that if you pull a null value out of a cuDF object with `__getitem__` you'll get `cudf.NA` as opposed to `None`. I could see that producing unexpected behavior in some workflows.

Reply:
Yea that's a bit of a scary change to push into 0.16 right before release. I 100% think it's the right direction but we should probably merge this into 0.17 then.